File: oc/Quartz2DDemo/Quartz2DDemo/WaterImageView.m
//
//  WaterImageView.m
//  Quartz2DDemo
//
//  Created by dfang on 2018-2-2.
//  Copyright © 2018年 east. All rights reserved.
//

#import "WaterImageView.h"

@implementation WaterImageView

+ (UIImage *)waterImageWithBgImageName:(NSString *)bgImageName waterImageName:(NSString *)waterImageName scale:(CGFloat)scale {
    UIImage *bgImage = [UIImage imageNamed:bgImageName];
    // 开始位图上下文
    UIGraphicsBeginImageContextWithOptions(bgImage.size, NO, scale);
    [bgImage drawInRect:CGRectMake(0, 0, bgImage.size.width, bgImage.size.height)];
    
    UIImage *waterImage = [UIImage imageNamed:waterImageName];
    // 水印图片缩放比例
    CGFloat waterScale = 0.4;
    CGFloat waterW = waterImage.size.width * waterScale;
    CGFloat waterH = waterImage.size.height * waterScale;
    CGFloat waterX = bgImage.size.width - waterW;
    CGFloat waterY = bgImage.size.height - waterH;
    [waterImage drawInRect:CGRectMake(waterX, waterY, waterW, waterH)];
    
    // 从位图上下文中获取当前编辑图片
    UIImage *newImage = UIGraphicsGetImageFromCurrentImageContext();
    // 结束当前编辑
    UIGraphicsEndImageContext();
    return newImage;
}

@end

File: swift/Quartz2DDemo/Quartz2DDemo/WaterImageView.swift
//
//  WaterImageView.swift
//  Quartz2DDemo
//
//  Created by dfang on 2018-2-4.
//  Copyright © 2018年 east. All rights reserved.
//

import UIKit

class WaterImageView: UIImage {
    
    class func waterImageWith(bgImageName:String, waterImageName:String, scale:Float) -> UIImage {
        let bgImage = UIImage.init(named: bgImageName)
        let waterImage = UIImage.init(named: waterImageName)
        let bgImageSize = (bgImage?.size)!
        UIGraphicsBeginImageContextWithOptions(bgImageSize, false, CGFloat(scale))
        bgImage?.draw(in: CGRect.init(origin: CGPoint.zero, size: bgImageSize))
        
        // 水印图片缩放比例
        let waterScale:CGFloat = 0.4
        let waterImageSize = (waterImage?.size)!
        let waterW = waterImageSize.width * waterScale
        let waterH = waterImageSize.height * waterScale
        let waterX = bgImageSize.width - waterW
        let waterY = bgImageSize.height - waterH
        waterImage?.draw(in: CGRect.init(x: waterX, y: waterY, width: waterW, height: waterH))
        let newImage = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()
        
        return newImage!
    }
}
